core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -675,16 +676,22 @@ public InodeDirectory getRoot() {
     // helper method for the shared logic.
     AlluxioURI path = inodePath.getUri();
     if (path.isRoot()) {
-      String errorMessage = ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path);
+      String errorMessage = "Not allowed to create existing root path: " + path;
       LOG.error(errorMessage);
       throw new FileAlreadyExistsException(errorMessage);
     }
     if (inodePath.fullPathExists()) {
       if (context instanceof CreateDirectoryContext
           && ((CreateDirectoryContext) context).getOptions().getAllowExists()) {
-        return new ArrayList<>();
+        return Collections.emptyList();
       } else {
-        throw new FileAlreadyExistsException(path);
+        String pathType = "file";
+        if (context instanceof CreateDirectoryContext) {
+          pathType = "directory";
+        }
+        String errorMessage =
+            String.format("Not allowed to create existing %s: %s", pathType, path);
```

Comment:
not clear to me. do we mean:
Not allowed to create a new file with existing path?

Reply:
I updated it. Does the message make more sense?